packages/workbox-build/src/templates/sw.js.tmpl

```diff
@@ -0,0 +1,33 @@
+<% if (importScripts) { %>
+importScripts(<%= importScripts.map(JSON.stringify).join(',') %>);
+<% } %>
+/**
+ * This service worker file needs to be registered by your web application.
+ * Please see https://developers.google.com/web/fundamentals/primers/service-workers/registration
+ */
+<% if(manifestEntries) {%>
+/**
+ * DO NOT EDIT __precacheManifest MANUALLY!
+ *
+ * The workboxSW.precache() method does the following:
+ * 1. Cache URLs in the manifest to a local cache.
+ * 2. When a network request is made for any of these URLs the response
+ *    will ALWAYS comes from the cache, NEVER the network.
+ * 3. When the service worker changes ONLY assets with a revision change are
+ *    updated, old cache entries are left as is.
+ *
+ * By changing the file manifest manually, your users may end up not receiving
+ * new versions of files because the revision hasn't changed.
+ *
+ * Please use workbox-build, workbox-cli, workbox-webpack-plugin, or some other
+ * equivalent build tool to generate the precache manifest for you as part of
+ * your pre-deployment build process.
+ */
+self.__precacheManifest = <%= JSON.stringify(manifestEntries, null, 2) %>.concat(self.__precacheManifest || []);
```

Comment:
Another change to help @goldhand is that we now check for `self.__precacheManifest`, which can be set inside of a script that is pulled in via `importScripts()`.

`self.__precacheManifest` and `manifestEntries` can be used together if you want some entries set via `importScripts()`, and others set via a parameter to the template.